app/modules/entities/components/layouts/homonym_deduplicates.svelte

```diff
@@ -25,19 +62,51 @@
         {@html icon('compress')}
         {I18n('merge homonyms')}
       </h4>
-      <!-- TODO: recover select all/unselect all/merge all selected buttons -->
-      <ul class="homonyms">
+
+      <div class="merge-homonyms-controls">
+        <button
+          on:click={selectAll}
+          aria-controls="selectable-homonyms"
+        >
+          {@html icon('check-square')}
+          <span class="button-label">{I18n('select all')}</span>
+          <span class="count">({homonyms.length})</span>
+        </button>
+        <button
+          on:click={unselectAll}
+          aria-controls="selectable-homonyms"
+        >
+          {@html icon('square')}
+          <span class="button-label">{I18n('unselect all')}</span>
+        </button>
+        <button
+          on:click={mergeSelectedSuggestions}
+          aria-controls="selectable-homonyms"
+          disabled={selectedHomonymsUris.length === 0}
+        >
+          {@html icon('compress')}
+          <span class="button-label">{I18n('merge selected suggestions')}</span>
+          <span class="count">({selectedHomonymsUris.length})</span>
+        </button>
+      </div>
+      <ul id="selectable-homonyms">
         {#each homonyms as homonym}
           {#if !homonym.merged}
             <li>
+              <input type="checkbox" bind:group={selectedHomonymsUris} value={homonym.uri} />
               <!-- TODO: recover list of subentities (typically author works) -->
-              <EntityListElement
+              <EntityListRow
                 entity={homonym}
-                actionType="merge"
                 parentEntity={entity}
-                noImageCredits="true"
                 displayUri="true"
               />
+              <MergeAction
+                bind:merge={homonym.merge}
+                fromEntityUri={homonym.uri}
+                targetEntityUri={entity.uri}
+                on:merged={() => homonym.merged = true}
+                on:isMerging={() => homonym.isMerging = true}
+              />
```

Comment:
If that works, we can keep it like that for now, but that could later be refactored to avoid the weirdness of exporting the merge function and calling it from somewhere under another name, which seems fragile. Also, the `merged` and `isMerging` flags could be set directly on the `homonym` object from the MergeAction component, no?

Reply:
This code is in the waiting for #372 which does perform de merge at the parent component level in order to merge sequentially 3 entities at a time (which was a feature parity).
So, I agree it looks weird but, its possible it would make sense in the context of #372 (and be reviewed at this time then?)